### PR TITLE
Release v2.9.0

### LINKED ISSUE #537: assert self.n_batch % self.nminibatches == 0  AssertionError
Hi I am trying to do a hyper parameter tuning by changing some of the default parameters of PPO2 algorithm. I have tried with some combinations already. 
My model
```
model = PPO2(MlpPolicy,env,gamma=0.99, n_steps=132, ent_coef=0.01,
             learning_rate=2.5e-2, vf_coef=0.5, max_grad_norm=0.5, lam=0.9,
             nminibatches=4, noptepochs=4, cliprange=0.1, cliprange_vf=None,
             verbose=1, tensorboard_log="./ppo_ahu_tensorboard/", _init_setup_model=True,
             policy_kwargs=None, full_tensorboard_log=False)

```
When I try to change the values of nminibatches and noptepochs, I always end up getting an error.
One such case, I tried with 
```
model = PPO2(MlpPolicy,env,gamma=0.8, n_steps=132, ent_coef=0.01,
             learning_rate=2.5e-2, vf_coef=0.5, max_grad_norm=0.5, lam=0.9,
             nminibatches=8, noptepochs=8, cliprange=0.1, cliprange_vf=None,
             verbose=1, tensorboard_log="./ppo_ahu_tensorboard/", _init_setup_model=True,
             policy_kwargs=None, full_tensorboard_log=False)

```
 and I get an error
```
assert self.n_batch % self.nminibatches == 0 AssertionError
```

What could be the reason? 
Thanks

### LINKED ISSUE #577: Weird tensorboard plots in GAIL
**Describe the bug**
While training GAIL, I get plots which go back in the x-axis in tensorboard. Please see example.

**Code example**
```python
from stable_baselines import GAIL
from stable_baselines.gail import ExpertDataset

dataset = ExpertDataset(expert_path='hopper.npz')
model = GAIL('MlpPolicy', 'Hopper-v2', dataset, tensorboard_log='logs/hopper-v2')
model.learn(total_timesteps=int(1e6))
```
Here is the graph that I get
Screenshot from 2019-11-23 14-18-54

**System Info**
Describe the characteristic of your environment:
 * Installed from source
 * Python version - 3.5.2
 * Tensorflow version - 1.14.0

This problem doesn't occur with TRPO where I get reasonable plots.

### LINKED ISSUE #610: Structure of ReplayBuffer._storage
In deepq.ReplayBuffer, the docstring for ```ReplayBuffer._storage``` says
```[(np.ndarray, float, float, np.ndarray, bool)]```
but shouldn't it actually be
```"""[(np.ndarray, [float], float, np.ndarray, bool)]```?

I think this would be correct for two reasons:
- Actions could be multi-dimensional
- The docstring for ```ReplayBuffer.add``` claims 
```:param action: ([float]) the action```, 
which I think is an inconsistency with the storage docstring.

Am I missing something here?

### LINKED ISSUE #625: The output of the making a gif example is wrong.
Hi,

**Describe the bug**
In the document of making a gif example. I found a bug and here is the code copied from the document.

```python
import imageio
import numpy as np

from stable_baselines import A2C

model = A2C("MlpPolicy", "LunarLander-v2").learn(100000)

images = []
obs = model.env.reset()
img = model.env.render(mode='rgb_array')
for i in range(350):
    images.append(img)
    action, _ = model.predict(obs)
    obs, _, _ ,_ = model.env.step(action)
    img = model.env.render(mode='rgb_array')

imageio.mimsave('lander_a2c.gif', [np.array(img[0]) for i, img in enumerate(images) if i%2 == 0], fps=29)
```
In the final line of the code, using img[0] cannot make a right GIF, and I think the enumerate operation is the key to this problem.

Details of the variables:
images: 350 * 400 * 600 * 3
img: 400 * 600 * 3
img[0]: 3

I've made a pr [link](https://github.com/hill-a/stable-baselines/pull/624)

**System Info**
Describe the characteristic of your environment:
 * Install the library using pip
 * GPU: titan xp
 * Python 3.5.6
 * TensorFlow-gpu=1.14.0

### PR DESCRIPTION
**Pull request by araffin**
## Description

This is the (Christmas) v2.9.0 release.

## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) for new features and bug fixes)

I also updated the documentation and docstrings at some places:
- addresses part of #625
- closes #610
- closes #577
- closes #537 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the [changelog](https://github.com/hill-a/stable-baselines/blob/master/docs/misc/changelog.rst) accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [x] I have ensured `pytest` and `pytype` both pass.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -19,4 +19,3 @@ filterwarnings =
 
 [pytype]
 inputs = stable_baselines
-python_version = 3.5
```
Comment (AdamGleave) [inline setup.cfg]: Are we dropping support for 3.5? If so we should specify the lowest supported version here. We should probably also add a `python_requires` in `setup.py`

Comment (araffin) [reply setup.cfg]: Is there a way to allow using pytype with python > 3.5?
I removed this line because I had to comment it each time I wanted to check typing, which was quite annoying.
If there is no other mean, I'll put that line back.

Comment (araffin) [reply setup.cfg]: so, no drop of python 3.5 support yet (I would have created an issue if this was the case)

Comment (AdamGleave) [reply setup.cfg]: I think `pytype` always type checks with the Python version of the interpreter it's running under, which is a bit annoying. If you don't specify it then it'll use the system default. It seems unfortunate to get different type check results depending on what system you're running on, but uncommenting the line does sound frustrating.

Could you install Python 3.5 in addition to the Python interpreter you normally use?

Comment (araffin) [reply setup.cfg]: I can but this would mean duplicating a complete conda env only for this particular purpose.
i'll put the line back
